app/models/claim/advocate_claim.rb

```diff
@@ -232,13 +236,16 @@ def set_supplier_number
 
     # create a blank fee for every basic fee type not passed to Claim::AdvocateClaim.new
     def instantiate_basic_fees
-      return unless new_record?
+      return unless case_type.present? && !case_type.is_fixed_fee?
+      return unless editable?
 
-      existing_basic_fee_type_ids = basic_fees.map(&:fee_type_id)
-      basic_fee_types = Fee::BasicFeeType.all
-      basic_fee_types.each do |basic_fee_type|
-        next if basic_fee_type.id.in?(existing_basic_fee_type_ids)
-        basic_fees << Fee::BasicFee.new_blank(self, basic_fee_type)
+      fee_type_ids = basic_fees.map(&:fee_type_id)
+      eligible_basic_fee_type_ids = eligible_basic_fee_types.map(&:id)
+      not_eligible_ids = fee_type_ids - eligible_basic_fee_type_ids
+      self.basic_fees = basic_fees.reject { |fee| not_eligible_ids.include?(fee.fee_type_id) }
+      eligible_basic_fee_types.each do |basic_fee_type|
+        next if fee_type_ids.include?(basic_fee_type.id)
+        basic_fees.build(fee_type: basic_fee_type, quantity: 0, amount: 0)
```

Comment:
The use of the Rails **conventions** to build a new basic fee for the claim instead of the existent `<< Fee::BasicFee.new_blank` makes **ALL** the different. I've spent hours debugging why I was getting duplicates with this! That's why! 😫